viamonvif/service.go

```diff
@@ -50,12 +63,18 @@ func (cfg *Config) Validate(_ string) ([]string, error) {
 	return []string{}, nil
 }
 
+type previewRequest struct {
+	rtspURL string
+}
+
 type rtspDiscovery struct {
 	resource.Named
 	resource.AlwaysRebuild
-	Credentials []device.Credentials
-	mdnsServer  *mdnsServer
-	logger      logging.Logger
+	Credentials        []device.Credentials
+	RTSPToSnapshotURIs map[string]string
+	mdnsServer         *mdnsServer
+	mu                 sync.RWMutex
```

Comment:
1. Let's start with just a normal mutex
2. Please use the[ mutex hat ](https://dmitri.shuralyov.com/idiomatic-go#mutex-hat)pattern 
3. make the fields private unless they need to be public

this lock only needs to protect `RTSPToSnapshotURIs` please move it over 
```suggestion
	Credentials        []device.Credentials
	
	rtspToSnapshotURIsMu  sync.Mutex
	rtspToSnapshotURIs map[string]string
	
	mdnsServer         *mdnsServer
```

Reply:
- Switched to normal write mutex
- Using hat pattern
- rtspToSnapshotURIs private